src/api/server/sync_io.rs

```diff
@@ -282,7 +282,8 @@ impl<F: FileSystem<S> + Sync, D: AsyncDrive, S: BitmapSlice> Server<F, D, S> {
     pub(super) fn rename2(&self, mut ctx: SrvContext<'_, F, D, S>) -> Result<usize> {
         let Rename2In { newdir, flags, .. } = ctx.r.read_obj().map_err(Error::DecodeMessage)?;
 
-        let flags = flags & (libc::RENAME_EXCHANGE | libc::RENAME_NOREPLACE) as u32;
+        let flags =
+            flags & (libc::RENAME_EXCHANGE | libc::RENAME_NOREPLACE | libc::RENAME_WHITEOUT) as u32;
```

Comment:
Even we accept the flag `RENAME_WHITEOUT`, the backend user-space file system daemon can handle the atomicity required by the flag?
I am afraid it is not suitable to accept the flag here. The flag only means to overlay itself rather than fuse.

Reply:
Yes, the atomicity should be handled by the underlying user-space FS. But take passthrough_fs as an example, it will just call `libc::SYS_renameat2` with the passthroughed flags and the syscall will ensure the atomicity. So, maybe just handing over the `RENAME_WHITEOUT` flag to the backend FS is fine. Thanks!

> https://github.com/cloud-hypervisor/fuse-backend-rs/blob/14926e5cf950574806ed2f78a79418f075ce260b/src/passthrough/sync_io.rs#L843